src/components/AssertionForm/useAssertionForm.ts

```diff
@@ -32,38 +43,164 @@ export function useAssertionForm() {
     address: USDC as Address,
     enabled: !!USDC,
   });
-  const currenciesData = { weth, dai, usdc };
+
+  const currenciesWithTokenData = useMemo(
```

Comment:
I'm memoizing a bunch of stuff because we're returning objects which sometimes need to be used in dependency arrays, which cause the hooks to run on every render. I've used the dev tools profiler to make sure we're only rendering when we need to.